docs/learn/dapp-developer/read-asset-data.md

```diff
@@ -266,6 +271,217 @@ console.log(assetMetadata);
 
 </Details>
 
+### Token ID Metadata
+
+To fetch the metadata of specific token IDs, you have to call the [`getDataForTokenId()`](../../contracts/contracts/LSP8IdentifiableDigitalAsset/extensions/LSP8Enumerable.md#getdatafortokenid) function of the LSP8 asset directly. Therefore, install a provider library to set up the contract and import the related contract ABIs:
+
+<Tabs groupId="provider-lib">
+  <TabItem value="ethers" label="ethers">
+
+```bash
+npm install ethers
+```
+
+```js
+// Add the necessary imports to your JS file
+import { ethers } from 'ethers';
+import lsp8Artifact from '@lukso/lsp-smart-contracts/artifacts/LSP8IdentifiableDigitalAsset.json';
+```
+
+  </TabItem>
+
+  <TabItem value="web3" label="web3">
+
+```bash
+npm install web3
+```
+
+```js
+// Add the necessary imports to your JS file
+import Web3 from 'web3';
+import lsp8Artifact from '@lukso/lsp-smart-contracts/artifacts/LSP8IdentifiableDigitalAsset.json';
+```
+
+  </TabItem>
+
+</Tabs>
+
+Afterwards, you can call the [`getDataForTokenId()`](../../contracts/contracts/LSP8IdentifiableDigitalAsset/extensions/LSP8Enumerable.md#getdatafortokenid) function using the related provider. As token IDs are only supported on LSP8, ensure that your contract address supports the [LSP8 Digital Identifiable Asset](../../standards/tokens/LSP8-Identifiable-Digital-Asset.md) standard. You can reuse the `isLSP8` variable from the [interface detection](#detect-the-contract-interface) detection above.
+
+:::caution
+
+Make sure to adjust `<myAssetAddress>` with the actual address of the asset. You can give it a try using an sample LSP8 asset on the testnet: [`0x8734600968c7e7193BB9B1b005677B4edBaDcD18`](https://wallet.universalprofile.cloud/asset/0x8734600968c7e7193BB9B1b005677B4edBaDcD18?network=testnet).
+
+:::
+
+<Tabs groupId="provider-lib">
+  <TabItem value="ethers" label="ethers">
+
+```js
+// ...
+
+if (isLSP8) {
+  const provider = new ethers.JsonRpcProvider(
+    'https://rpc.testnet.lukso.gateway.fm',
+  );
+
+  // Token ID as Bytes32 value (1)
+  const tokenID =
+    '0x0000000000000000000000000000000000000000000000000000000000000001';
+
+  // Create contract instance
+  const myAssetContract = new ethers.Contract(
+    '<myAssetAddress>',
+    lsp8Artifact.abi,
+    provider,
+  );
+
+  // Get the encoded asset metadata
+  const tokenIdMetadata = await myAssetContract.getDataForTokenId(
+    tokenID,
+    ERC725YDataKeys.LSP4['LSP4Metadata'],
+  );
+
+  const erc725js = new ERC725(lsp4Schema);
+
+  // Decode the metadata
+  const decodedMetadata = erc725js.decodeData([
+    {
+      keyName: 'LSP4Metadata',
+      value: tokenIdMetadata,
+    },
+  ]);
+  console.log(decodedMetadata);
+}
+```
+
+  </TabItem>
+
+  <TabItem value="web3" label="web3">
+
+```js
+// ...
+
+if (isLSP8) {
```

Comment:
I would suggest not wrapping the whole snipped in this `if (isLSP8)`. Its not really relevant to the guide imo as this guide only applies to LSP8

Reply:
okay, maybe early return of something would clean it up a bit